From da23d09f215c83e20a281484fdbc171341ebc9ec Mon Sep 17 00:00:00 2001
From: Rob Clark <robdclark@gmail.com>
Date: Thu, 25 Apr 2013 17:38:24 -0400
Subject: [PATCH] kgsl: drm: remove checking on 'active'

We are using this field as an evil way to synchronize between mesa and
ddx, and the error checking gets in the way.. so get rid of it.
---
 drivers/gpu/msm/kgsl_drm.c | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/drivers/gpu/msm/kgsl_drm.c b/drivers/gpu/msm/kgsl_drm.c
index e3f6f3b219b8a..e7ea2235b0123 100644
--- a/drivers/gpu/msm/kgsl_drm.c
+++ b/drivers/gpu/msm/kgsl_drm.c
@@ -817,11 +817,6 @@ kgsl_gem_get_bufinfo_ioctl(struct drm_device *dev, void *data,
 	mutex_lock(&dev->struct_mutex);
 	priv = obj->driver_private;
 
-	if (!kgsl_gem_memory_allocated(obj)) {
-		DRM_ERROR("Memory not allocated for this object\n");
-		goto out;
-	}
-
 	for (index = 0; index < priv->bufcount; index++) {
 		args->offset[index] = priv->bufs[index].offset;
 		args->gpuaddr[index] = priv->bufs[index].gpuaddr;
@@ -832,7 +827,6 @@ kgsl_gem_get_bufinfo_ioctl(struct drm_device *dev, void *data,
 
 	ret = 0;
 
-out:
 	drm_gem_object_unreference(obj);
 	mutex_unlock(&dev->struct_mutex);
 
@@ -899,15 +893,9 @@ kgsl_gem_set_active_ioctl(struct drm_device *dev, void *data,
 	mutex_lock(&dev->struct_mutex);
 	priv = obj->driver_private;
 
-	if (args->active < 0 || args->active >= priv->bufcount) {
-		DRM_ERROR("Invalid active buffer %d\n", args->active);
-		goto out;
-	}
-
 	priv->active = args->active;
 	ret = 0;
 
-out:
 	drm_gem_object_unreference(obj);
 	mutex_unlock(&dev->struct_mutex);
 
